### PR TITLE
Story 346 unittest for fees and subscriptions

### PR DESCRIPTION
**Pull request by johevemi**
## Description
Split subscription and fee calculator methods op into small more testable methods and create unit test for those methods 

## References

* #346

### CODE REVIEW

**Review by jonasdmoeller — APPROVED** [review]

File: source/databricks/tests/wholesale_utils/calculators/test_fee_calculators.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    fee_charges = spark.createDataFrame(fee_charges, schema=charges_flex_settled_consumption_schema)  # fee_charges and charges_flex_settled_consumption has the same schema
+
+    # Act
+    result = filter_on_metering_point_type_and_settlement_method(fee_charges)
+
+    # Assert
+    assert result.count() == expected
+
+
+charges_flex_settled_consumption_dataset_1 = [("chargea-D01-001", "chargea", "D01", "001", Decimal("100.10"), datetime(2020, 1, 1, 0, 0), "E17", "D01", "chargea", 1, 1)]
```
Comment (mknic) [inline source/databricks/tests/wholesale_utils/calculators/test_fee_calculators.py:139]: small typo in `chargea...`. should be `charges...`. I think it's applicable to more of the tests.

File: source/databricks/tests/wholesale_utils/calculators/test_subscription_calculators.py [hunk 2]
```diff
@@ -167,3 +195,72 @@ def test__calculate_daily_subscription_price__charge_price_change_with_two_diffe
 
     # Assert
     assert result.collect() == expected.collect()
+
+
+subscription_charges_dataset_1 = [("chargea-D01-001", "chargea", "D01", "001", Decimal("200.50"), datetime(2020, 1, 1, 0, 0), "E17", "D01", "chargea", 1, 1)]
```
Comment (mknic) [inline source/databricks/tests/wholesale_utils/calculators/test_subscription_calculators.py:200]: same issue here

File: source/databricks/tests/wholesale_utils/calculators/test_subscription_calculators.py [hunk 3]
```diff
@@ -167,3 +195,72 @@ def test__calculate_daily_subscription_price__charge_price_change_with_two_diffe
...
+    subscription_charges = spark.createDataFrame(subscription_charges, schema=charges_flex_settled_consumption_schema)  # subscription_charges and charges_flex_settled_consumption has the same schema
+
+    # Act
+    result = filter_on_metering_point_type_and_settlement_method(subscription_charges)
+
+    # Assert
+    assert result.count() == expected
+
+
+charges_flex_settled_consumption_dataset_1 = [("chargea-D01-001", "chargea", "D01", "001", Decimal("100.10"), datetime(2020, 1, 1, 0, 0), "E17", "D01", "chargea", 1, 1)]
```
Comment (mknic) [inline source/databricks/tests/wholesale_utils/calculators/test_subscription_calculators.py:223]: another one. a search and replace will fix it I guess :-)

**Review by mknic — CHANGES_REQUESTED** [review]
Very nice work. Besides a small typo in multiple places it looks good.

File: source/databricks/tests/wholesale_utils/calculators/test_fee_calculators.py (hunk 1 above)
Comment (johevemi) [reply source/databricks/tests/wholesale_utils/calculators/test_fee_calculators.py:139]: It is actually not a typo but is meant to be read as "charge a, charge b, charge c" to create a charge_id. But I see your point and will change it to "001" or something like that

File: source/databricks/tests/helpers/dataframe_defaults.py [hunk 4]
```diff
@@ -13,15 +13,14 @@
...
 from geh_stream.schemas import charges_schema, charge_links_schema, charge_prices_schema, metering_point_schema, market_roles_schema
-from geh_stream.codelists import ResolutionDuration
 import pytest
 import pandas as pd
 
 
-const_charge_id = "chargea"
-const_charge_type = "D01"
+const_charge_id = "001"
+const_charge_type = ChargeType.subscription
```
Comment (ZavezX) [inline source/databricks/tests/helpers/dataframe_defaults.py:23]: Why set default charge type? Isn't it supposed to be used for all types of charges?

**Review by ZavezX — APPROVED** [review]
Just a little thought, otherwise it looks good.

**Pull request merged**